pydy/viz/scene.py

```diff
@@ -350,16 +497,15 @@ def generate_visualization_json_system(self, system, **kwargs):
         ``generate_visualization_json`` method.
 
         """
-        if not isinstance(system, System):
-            msg = "{} should be a valid pydy.System object".format(system)
-            raise TypeError(msg)
-        else:
-            self._system = system
 
-        self.generate_visualization_json(system.states,
-                                         system.constants.keys(),
-                                         system.integrate(),
-                                         system.constants.values(), **kwargs)
+        self.system = system
+        prefix = None
+        for k, v in kwargs.items():
+            if k == 'fps':
+                self.frames_per_second = v
+            if k == 'outfile_prefix':
+                prefix = v
+        self._generate_json(prefix=prefix)
 
     def create_static_html(self, overwrite=False, silent=False):
```

Comment:
There definitely need to be a call to `_generate_json()` in this method, otherwise there are no `self._scene_json_file`, etc defined.